### ISSUE TITLE
borg check --repair does not repair object count mismatch

### ISSUE THREAD

**Issue by ThomasWaldmann** [borgbackup/borg#1396]
Tried to repair the broken repo from #1395, but couldn't:

```
(borg-env)tw@tux:~/w/borg$ borg check -v repo
Starting repository check
Completed repository check, no problems found.
Starting archive consistency check...
Analyzing archive 3.checkpoint (3/3)
Analyzing archive 2.checkpoint (2/3)
Analyzing archive 1.checkpoint (1/3)
Archive consistency check complete, no problems found.
```
```
(borg-env)tw@tux:~/w/borg$ borg check -v --repair repo
'check --repair' is an experimental feature that might result in data loss.
Type 'YES' if you understand this and want to continue: YES
Starting repository check
Completed repository check, no problems found.
Starting archive consistency check...
Analyzing archive 3.checkpoint (3/3)
Analyzing archive 2.checkpoint (2/3)
Analyzing archive 1.checkpoint (1/3)
Archive consistency check complete, no problems found.
```

```
(borg-env)tw@tux:~/w/borg$ borg check -v repo
Starting repository check
Index object count mismatch. 92 != 93
Completed repository check, errors found.
```
```
(borg-env)tw@tux:~/w/borg$ borg check -v --repair repo
'check --repair' is an experimental feature that might result in data loss.
Type 'YES' if you understand this and want to continue: YES
Starting repository check
Completed repository check, no problems found.
Starting archive consistency check...
Analyzing archive 3.checkpoint (3/3)
Analyzing archive 2.checkpoint (2/3)
Analyzing archive 1.checkpoint (1/3)
1 orphaned objects found!
Archive consistency check complete, problems found.
```
```
(borg-env)tw@tux:~/w/borg$ borg check -v --repair repo
'check --repair' is an experimental feature that might result in data loss.
Type 'YES' if you understand this and want to continue: YES
Starting repository check
Completed repository check, no problems found.
Starting archive consistency check...
Analyzing archive 3.checkpoint (3/3)
Analyzing archive 2.checkpoint (2/3)
Analyzing archive 1.checkpoint (1/3)
1 orphaned objects found!
Archive consistency check complete, problems found.
```
```
(borg-env)tw@tux:~/w/borg$ borg check -v repo
Starting repository check
Index object count mismatch. 92 != 93
Completed repository check, errors found.
```

**Comment by enkore** [comment]
What I don't get here is that the index mismatch is a repository-level error. This really shouldn't happen.

**Comment by enkore** [comment]
A bug in compact_segments.

This is *not* critical. **It does not lead to data loss**. In fact, quite the opposite. Borg stumbles over supposedly-deleted data, we just "forgot" the fact that we deleted it ;)

More in-depth analysis and fix will be coming.

**Comment by ThomasWaldmann** [comment]
@enkore so, is this fixed by #1448?

**Comment by enkore** [comment]
No, 86d95fe is the draft-fix
